support/BRFileService.c

```diff
@@ -30,10 +30,75 @@
 #include <dirent.h>
 #include <sys/stat.h>
 #include <errno.h>
+#include <sqlite3.h>
```

Comment:
Need to think about this; do we want iOS and Android using different versions of sqlite? Would we prefer that they use the same (included) version, at the cost of binary size?

Reply:
I think IOS uses the built-in, 'blessed-by-Apple' version.  And Android should use the 'blessed-by-Google' version.  I certainly don't want to deal with upgrades and issue tracking on a version we choose.

Reply:
SQLite is not exposed by the Android NDK. It is only available (assuming you are "following the rules") in the SDK (i.e. at the Java layer).

So, 'blessed-by-Google' version does not exist.

Reply:
So what should we do?  Using SQLite3 was a test for performance of BRFileService; Adrian reports that there are no longer any slow ups. 

Do we have another non-sqlite option?  (Maintain a Set of {blocks, peers, transaction}; on any change, write the entire set out to one file (for each type) - like in plist or json format?)

If you include the Android SDK are the .o and .h unavailable (deep in the JAR/AAR-thingy)?

Reply:
Maybe but I don't think we want to be doing that. We could like just dlopen the library if we wanted to get hacky.

I would think we would consider a) including SQLite or b) use callbacks to expose the platform's SQLite implmentation.

Reply:
Should we bundle sqlite with Core?

Reply:
Of the options, I'd say so. Assuming everyone is good with that, from a licensing/size/maintenance point of view.

As we've discussed, the file service could potentially be made to use callbacks so the platform sqlite libraries could be used. There are issues with this (possibly performance on Java, library versioning/behaviour, etc.).

We could also hedge a bit here and A) do the callback approach but B) also bundle in sqlite, with callbacks being provided for the bundled sqlite. Would give us flexibility, as well as (potentially, if we want to, needs more thought) use the platform sqlite instance where available (i.e. iOS) and the bundled one where not (easily i.e. Android).